packages/server/src/commands/export/domain.ts

```diff
@@ -21,9 +21,24 @@ export interface ExportFile {
   content: string
 }
 
-export interface MetaExportFiles {
-  "phero-manifest.d.ts": string
-  "phero-execution.js": string
-  "phero.js": string
-  "package.json": string
+export enum MetaExportLockFileName {
+  Npm = "package-lock.json",
+  Yarn = "yarn.lock",
+  Pnpm = "pnpm-lock.yaml",
+}
+
+export interface MetaExportLockFile {
+  name: MetaExportLockFileName
+  path: string
+}
+
+type MetaExportFileContent = string
+
+export type MetaExportFiles = {
+  "phero-manifest.d.ts": MetaExportFileContent
+  "phero-execution.js": MetaExportFileContent
+  "phero.js": MetaExportFileContent
+  "package.json": MetaExportFileContent
+} & {
+  [key in MetaExportLockFileName]?: MetaExportFileContent
```

Comment:
If we specify it like this we can either have all lock-files or none at all. Maybe we can do something like below? It's a little more type safety than we need right now, but it might save us some mistakes in the future.
```
export type MetaExportFiles = {
  "phero-manifest.d.ts": MetaExportFileContent
  "phero-execution.js": MetaExportFileContent
  "phero.js": MetaExportFileContent
  "package.json": MetaExportFileContent
} & (
  | {
      [MetaExportLockFileName.Npm]: MetaExportFileContent
    }
  | {
      [MetaExportLockFileName.Yarn]: MetaExportFileContent
    }
  | {
      [MetaExportLockFileName.Pnpm]: MetaExportFileContent
    }
)
```

Reply:
This was my initial thought as well, but couldn't get it to work properly. Your comment pushed me to try again and got it fixed in https://github.com/phero-hq/phero/pull/86/commits/267df921ea0ef980045fd05f2c85a8796cfe557d 😄